### ISSUE TITLE
Investigate Kubernetes + Docker as architecture for SaaS

### ISSUE THREAD

**Issue by deruelle** [Mobicents/RestComm#415]
Idea of using Kubernetes and Docker Containers to manage the SaaS and spin up Docker Containers per customer based on load of each customer to guarantee isolation. Potentially bundle containers for multiple customers on the same machine and be able to move them based on the growing load

**Comment by gsaslis** [comment]
Is this still an ongoing effort?

@deruelle out of curiosity, why did you substitute kubernetes with mesos in the title?

the 3 main orchestrators at this point in time are kubernetes, docker swarm and apache mesos. is there already any decision / focus towards one of these?

**Comment by deruelle** [comment]
@gsaslis yes this is still an ongoing effort. We still need to evaluate and test which one will fit best. There is also OpenShift to consider.

Prior to this effort we are currently in the process of separating a number of services bundled with Restcomm Connect such as the Visual Designer (RVD), Olympus, Dashboard, CDR, Registrar, ... 

RVD is now almost separated and it will be time to experiment with one of those orchestrator. Would you like to put up a proof of concept or discuss that more deeply ?

**Comment by gsaslis** [comment]
yes, I consider openshift more of an opinionated kubernetes deployment, rather than an orchestrator itself, which is why I didn't mention it, but yes, we could definitely focus on that as our deployment platform!

about RVD in particular, care to point me in the right direction?

Is the https://github.com/RestComm/visual-designer/ repo the place to be looking for RVD ?

I see there's an open issue https://github.com/RestComm/visual-designer/issues/146 - should that be done first, before beginning to work on this PoC ?

(maybe we should switch to Gitter for a more detailed discussion on this? )

**Comment by deruelle** [comment]
@gsaslis yes https://github.com/RestComm/visual-designer/  is the right location.

You're correct on RestComm/visual-designer#146 @otsakir can give you some insights on where we stand on this.
